### PR TITLE
Add documentation regarding metering setup on a new zone

### PR DESCRIPTION
**Pull request by simu**
## Checklist

- [x] Try to isolate changes into separate PRs (to build a better changelog).
- [x] Categorize the PR by setting a good title and adding one of the labels:
      `change`, `decision`, `requirement/quality`, `requirement/functional`, `dependency`
      as they show up in the changelog

### CODE REVIEW

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc [hunk 1]
```diff
@@ -117,3 +117,62 @@ parameters:
 
 . Compile and push the cluster catalog
 . Wait for Argo CD to sync the config
+
+== Configure metering metrics collection
+
+We collect a small set of metrics which are used for metering from each zone.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/zone-setup.adoc:123]:
```suggestion
We collect a small set of metrics which are used for metering from each {zone}.
```

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc [hunk 2]
```diff
@@ -117,3 +117,62 @@ parameters:
...
+This is done by deploying a https://thanos.io/tip/components/receive.md/[Thanos Receiver] on each zone and configure the zone's cluster monitoring to send selected metrics to the Thanos Receiver via Prometheus's remote write feature.
+
+. Create an S3 bucket which can be accessed using the same credentials as the cluster's registry bucket.
+Use name `${CLUSTER_ID}-appuio-metering` for the bucket.
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: small name suggestion. `metering` is for me a device or component to measure something. However the metrics we put into the bucket are already measured :)
```suggestion
Use name `${CLUSTER_ID}-appuio-metrics` for the bucket.
```

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc [hunk 3]
```diff
@@ -117,3 +117,62 @@ parameters:
...
+    commonConfig:
+      volumeClaimTemplate:
+        spec:
+          storageClassName: ssd
+      # Override default security context for all components
+      securityContext: {}
+    objectStorageConfig:
+      type: S3
+      config:
+        bucket: c-cluster-id-1234-appuio-metering <3>
```
Comment (ccremer) [inline docs/modules/ROOT/pages/how-to/zone-setup.adoc:158]:
```suggestion
        bucket: c-cluster-id-1234-appuio-metrics <3>
```

**Review by ccremer — APPROVED** [review]

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc (hunk 2 above)
Comment (simu) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: I've used metering everywhere, since that's the term which is used in the stories.

Comment (ccremer) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: Well, we use both terms :shrug: 
See e.g. https://kb.vshn.ch/appuio-cloud/references/architecture/metering.html

But I don't think we should be using `metrics` and `metering` interchangeably (as synonyms). For me, metering is the whole process, whereas metrics are just the data part of metering.

@corvus-ch Any opinions?

Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: For me the term `metering` refers to metrics that are used for statistics and invoicing. `metrics` is more bound to metrics used for monitoring and alerting. That is just my take on it. Whatever naming we choose, we must be able to distinguish between those cases. @mhutter is working on a centralized monitoring that builds on the same tools. Those must remain separated setups and their naming should make it clear.

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc [hunk 4]
```diff
@@ -117,3 +117,62 @@ parameters:
...
+
+== Configure metering metrics collection
+
+We collect a small set of metrics which are used for metering from each zone.
+This is done by deploying a https://thanos.io/tip/components/receive.md/[Thanos Receiver] on each zone and configure the zone's cluster monitoring to send selected metrics to the Thanos Receiver via Prometheus's remote write feature.
+
+. Create an S3 bucket which can be accessed using the same credentials as the cluster's registry bucket.
+Use name `${CLUSTER_ID}-appuio-metering` for the bucket.
+
+. Add configuration
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-to/zone-setup.adoc:129]: Is there a reason to not add this to the zone configuration?

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc [hunk 5]
```diff
@@ -117,3 +117,62 @@ parameters:
...
+        remoteWrite:
+          - name: appuio-metering
+            url: http://thanos-receive.syn-thanos.svc:19291/api/v1/receive
+            writeRelabelConfigs:
+              - sourceLabels: ['__name__']
+                regex: '(container_memory_usage_bytes|kube_pod_container_resource_requests|kube_persistentvolumeclaim_resource_requests_storage_bytes)' <1>
+                action: keep
+
+  thanos:
+    cluster_kubernetes_version: '1.22' <2>
```
Comment (corvus-ch) [inline docs/modules/ROOT/pages/how-to/zone-setup.adoc:148]: Why not use dynamic facts?

**Review by corvus-ch — COMMENTED** [review]
I am probably miss some context but this feels to add manual steps which could be covered through the zone configuration in our hierarchy.

As for the naming, `metering` seems to be the way to go.

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc (hunk 2 above)
Comment (ccremer) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: Hm, if metrics is only for monitoring and alerting, wouldn't that be inconsistent with https://kb.vshn.ch/appuio-cloud/references/architecture/metrics-of-interest.html and ff?
Maybe we need to better define those terms in a separate PR.
Let's go with the `appuio-metering` bucket name for now, it's not that much of an issue for me.

Comment (corvus-ch) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: I agree. And yes, this is bottering me too.

Comment (simu) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:127]: I think we're confusing two levels of abstraction here. On one level of abstraction we have metrics which are relevant for invoicing (what I've been calling "metering metrics" or similar in commit messages and config comments), on the other more abstract level we have the "APPUiO Metering" component (for lack of a better name).

To make it clear to which component this bucket belongs, I've been using the `appuio-metering` suffix to indicate the system component which uses this bucket.

In any case, I agree that we should probably clearly define terms around metrics and metering, but that's clearly out of scope for this PR.

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc (hunk 4 above)
Comment (simu) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:129]: It is part of the zone configuration, but this how-to is the generic one which doesn't assume that you've got access to the VSHN global defaults (as far as I understand).

File: docs/modules/ROOT/pages/how-to/zone-setup.adoc [hunk 6]
```diff
@@ -117,3 +117,62 @@ parameters:
...
+----
+applications:
+  - thanos
+parameters:
+  openshift4_monitoring:
+    configs:
+      prometheusK8s:
+        remoteWrite:
+          - name: appuio-metering
+            url: http://thanos-receive.syn-thanos.svc:19291/api/v1/receive
```
[the remark this answers is not in the document]
Comment (simu) [reply docs/modules/ROOT/pages/how-to/zone-setup.adoc:141]: This will most likely be renamed as soon as component-thanos multi-instance support is release

**Review by corvus-ch — APPROVED** [review]

**Pull request merged**